Doc/library/typing.rst

```diff
@@ -1409,8 +1410,74 @@ These can be used as types in annotations. They all support subscription using
               ...
 
    Sometimes it would be convenient to use a user-defined boolean function
-   as a type guard.  Such a function should use ``TypeGuard[...]`` as its
-   return type to alert static type checkers to this intention.
+   as a type predicate.  Such a function should use ``TypeIs[...]`` or
+   :data:`TypeGuard` as its return type to alert static type checkers to
+   this intention.  ``TypeIs`` usually has more intuitive behavior than
+   ``TypeGuard``, but it cannot be used when the input and output types
+   are incompatible (e.g., ``list[object]`` to ``list[int]``) and when the
+   function does not return ``True`` for all instances of the narrowed type.
+
+   Using  ``-> TypeIs`` tells the static type checker that for a given
```

Comment:
Would it be clearer to use `TypeIs[...]` or `TypeIs[SomeType]` here? The latter would also give you a name that might be clearer/shorter than "the type inside TypeIs" below.